### PR TITLE
Promisy fun

### PR DESCRIPTION
**Pull request by jdungan**
Another refactor to speed up load time.  It work but comes with a recursive promise chain -- looking forward to a good discussion before merging.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,3 +37,23 @@ combined tasks:
...
+* **build-chrome-directory**: create chrome dev directroy in the chrome.source directory with dev settings
+
+#### workon-{browser}-directory or workon-{browser}-directory:
+watches extension files and rebuilds
+
+
+#### publish-{browser}-file
+* **publish-firefox-file**: create xpi for FF prod
+* **publish-chrome-file**: create zip for chrome and opera
+* **publish-all**: creates all browser extension files
```
Comment (groovecoder) [inline README.md]: When I try to run any of these `gulp` tasks, I get:
```
Error: Task requires a name that is a string
    at Gulp.Orchestrator.add (/Users/lcrouch/code/codesy/widgets/node_modules/orchestrator/index.js:41:10)
    at Object.<anonymous> (/Users/lcrouch/code/codesy/widgets/gulpfile.js:146:10)
    at Module._compile (module.js:573:32)
    at Object.Module._extensions..js (module.js:582:10)
    at Module.load (module.js:490:32)
    at tryModuleLoad (module.js:449:12)
    at Function.Module._load (module.js:441:3)
    at Module.require (module.js:500:17)
    at require (internal/module.js:20:19)
    at Liftoff.handleArguments (/usr/local/lib/node_modules/gulp/bin/gulp.js:116:3)
```

File: src/issue.js [hunk 2]
```diff
@@ -1,71 +1,75 @@
...
-                }
-            },
-            iframe:
-                { attr:{
-                    id: "codesy_iframe",
-                    style: "visibility: collapse;",
-                    scrolling: "no",
-                    seamless: "seamless"
-                }
+const make_widget_appender = (domain) => {
```
Comment (jdungan) [inline src/issue.js]: artistic license?  Actually, I'm settling on named functions if it longer than one line.

File: README.md (hunk 1 above)
Comment (jdungan) [inline README.md]: d580d3a should fix this.

**Comment by groovecoder** [comment]
Still getting errors during gulp ...
```
groovecoder:widgets lcrouch$ gulp workon-firefox-directory

/Users/lcrouch/code/codesy/widgets/gulpfile.js:44
javascript_src = function({source, destination}) {
                          ^
SyntaxError: Unexpected token {
    at Module._compile (module.js:439:25)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Liftoff.handleArguments (/usr/local/lib/node_modules/gulp/bin/gulp.js:116:3)
    at Liftoff.<anonymous> (/usr/local/lib/node_modules/gulp/node_modules/liftoff/index.js:181:16)
    at module.exports (/usr/local/lib/node_modules/gulp/node_modules/liftoff/node_modules/flagged-respawn/index.js:17:3)
    at Liftoff.<anonymous> (/usr/local/lib/node_modules/gulp/node_modules/liftoff/index.js:174:9)
groovecoder:widgets lcrouch$ gulp workon-chrome-directory

/Users/lcrouch/code/codesy/widgets/gulpfile.js:44
javascript_src = function({source, destination}) {
                          ^
SyntaxError: Unexpected token {
    at Module._compile (module.js:439:25)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Liftoff.handleArguments (/usr/local/lib/node_modules/gulp/bin/gulp.js:116:3)
    at Liftoff.<anonymous> (/usr/local/lib/node_modules/gulp/node_modules/liftoff/index.js:181:16)
    at module.exports (/usr/local/lib/node_modules/gulp/node_modules/liftoff/node_modules/flagged-respawn/index.js:17:3)
    at Liftoff.<anonymous> (/usr/local/lib/node_modules/gulp/node_modules/liftoff/index.js:174:9)
groovecoder:widgets lcrouch$
```

File: README.md [hunk 3]
```diff
@@ -10,15 +10,7 @@ The codesy.io widget is an add-on for Firefox, Chrome, and Opera that adds codes
...
-3. Run these gulp tasks to watch changes to files and compile extensions.
-  * gulp dev-chrome-unpacked - creates chrome/ directory and watches for changes
-  * gulp dev-chrome-packed - creates a zip file in /chrome and watches for changes
-  * gulp dev-firefox-unpacked - creates firefox/ directory and watches for changes
-  * gulp dev-firefox-packed - creates an xpi in firefox/ file and watches for changes
-
-combined tasks:
-  * gulp dev-packed - creates addon packages and watches for changes
-  * gulp dev-unpacked - creates directories with addon files and watches all for changes
+3. See the list of gulp tasks build or watch changes to files and compile extensions:
```
Comment (jdungan) [inline README.md]: Does 382d507 fix this?

**Comment by jdungan** [comment]
hmm? gulp workon-chrome-directory is working for me.

File: src/on_install.js [hunk 4]
```diff
@@ -19,24 +19,24 @@ const wait_for = (id, resolve)=>{
...
 }
 
-const select_them = (tabs)=> {
+function select_them (tabs) {
     tabs.map( ({id}) => chrome.tabs.update(id, {selected:true}) )
     return tabs
 }
 
-when_installed = ({reason}) => {
+when_installed ({reason}) => {
```
Comment (groovecoder) [inline src/on_install.js]: needs to be:
```
function when_installed({reason}) {
...
}
```

**Pull request merged**